### DISCUSSION TITLE
Issues with ImageArrayRegion

### DISCUSSION THREAD

**Discussion by sousvideonlow** [Spawnova/ShinsImageScanClass#5] [Q&A] [answered]
Hi I have no issues with the rest but only with ImageArrayRegion

```
msgbox,% "image = " scan.image("zoom.png") ; returns 1
msgbox,% "imageregion = " scan.ImageRegion("zoom.png",0,0,3000,3000) ; returns 1
imgarrayregion:={}
scan.ImageArrayRegion("zoom.png",imgarrayregion,0,0,3000,3000)
for i,v in imgarrayregion
{
    msgbox,% i a_tab v ; returns blanks
}
msgbox,% obj2string(imgarrayregion)
```

**Comment by sousvideonlow** [comment, marked as answer]
Ok I think I might have fixed it. On line 262 of your class

I changed it to 

```
count := DllCall(this._ScanImageArrayRegion,"Ptr",this.dataPtr,"Ptr",this.imageCache[image],"uint",x1,"uint",y1,"uint",w,"uint",h,"uchar",variance,"uchar",centerResults,"int")
```

**Reply by Spawnova** [reply]
Ah! Embarrasing, When I was writing the ahk functions I copied ImageArray and edited it to fit ImageArrayRegion, but forgot to change the dllcall structure!
Sorry about that! Thank you for bringing it to my attention though I will update the class file with this change =P

**Reply by sousvideonlow** [reply]
very nice. thanks for maintaining it. I love your class. I was previously using findtext, but yours seems just slightly faster for large image. Find text is great for small areas of text though.

**Reply by Spawnova** [reply]
No problem, thanks for the kind words =)
I check here daily for any issues or suggestions, so as long as people keep posting, I'll keep updating lol
Thanks again for finding that error!
